src/index.js

```diff
@@ -190,12 +190,22 @@ const ruleFunction = (properties, options, context = {}) => (root, result) => {
 
       // test variable
       if (ignoreVariables) {
-        validVar = reVar.test(value) || cssLoaderValues.test(value)
+        // @TODO: deviant regexes to primary options need to be evaluated
```

Comment:
This means config where `Regex`'s are not equal in primary and secondary options, like:

```js
// .stylelintrc
"rules": {
  // ...
  "scale-unlimited/declaration-strict-value": ["/color$/", "margin", {
    ignoreFunctions: {
      "/border-*-color$/": false,
    },
  }],
  // ...
}
```